lib/policy-download.js

```diff
@@ -51,34 +51,49 @@ PolicyDownload.prototype.considerOneKey = function(key){
 
     var s3 = self.aws.configureS3();
     s3.headObject({Bucket: self.settings.s3_bucket,Key: key},function(err,result){
-        if (err) return self.recordError(err);
+
+        function logError(err){
+            self.logger.error('download error: ' + err + ' ' + _.last(key.split("/")));
+            self.context.action = self.context.action ? self.context.action + '+error' : 'error';
+            self.context.error = err;
+            return _.defer(self.considerNextKeyCallback);
+        }
+
+        if (err) return logError(err);
 
         var keyParts = key.split('/');
         var filename = keyParts[keyParts.length - 1];
         var lastTimestamp = self.lastTimestamps[key];
         var targetFilename = self.makeTargetFilename(filename);
+        var s3FileSizeInBytes = parseInt(result.ContentLength);
 
         self.filesToDelete = _.difference(self.filesToDelete,[filename]);
 
-        if (self.helpers.fileExists(targetFilename) && lastTimestamp && result.LastModified == lastTimestamp) {
+        if (self.helpers.fileExists(targetFilename) && (!lastTimestamp || result.LastModified === lastTimestamp)) {
             self.logger.debug('...already downloaded');
             self.context.result.unchanged++;
+            self.lastTimestamps[key] = result.LastModified;
             return _.defer(self.considerNextKeyCallback);
         }
 
         s3.getObject({Bucket: self.settings.s3_bucket,Key: key},function(err,data){
-            if (err) return self.recordError(err);
-
-            self.helpers.writeFile(targetFilename,data.Body,function(err){
-                if (err) return self.recordError(err);
+            if (err) return logError(err);
 
-                self.logger.message('...downloaded: ' + targetFilename);
+            var tempFilename = targetFilename + '.tmp';
+            self.helpers.writeFile(tempFilename, data.Body, function(err){
+                if (err) return logError(err);
 
-                self.context.result.added++;
+                if (self.helpers.fileSizeInBytes(tempFilename) !== s3FileSizeInBytes) {
+                    self.helpers.unlinkSync(self.makeTargetFilename(filename));
```

Comment:
let's count this as an error by calling "logError(...)" ... right now it is silent when this condition occurs